logstash-core/lib/logstash/plugins/registry.rb

```diff
@@ -0,0 +1,83 @@
+# encoding: utf-8
+require 'singleton'
+require "rubygems/package"
+
+module LogStash
+  class Registry
+
+    ##
+    # Placeholder class for registered plugins
+    ##
+    class Plugin
```

Comment:
Using a class here seems heavy. Is there a way we can do this just by adding these as class methods to `LogStash::Registry`?